packages/sdk/src/types/userGroup.ts

```diff
@@ -0,0 +1,48 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+let lastIndex = 0;
+const groupIndex: { [name: string]: number } = {};
+
+/**
+ * A set of user group IDs. User groups are used to selectively enable several different
+ * properties of [[Actor]]s based on the memberships of the viewing [[User]]. See [[User.groups]],
+ * [[Appearance.enabled]], and [[Collider.enabled]].
+ */
+export default class UserGroup extends Set<string> {
+    constructor(initialContents: string[] = []) {
+        super(initialContents);
+    }
+
+    public add(item: string): this {
+        if (!groupIndex[item]) {
+            // according to MDN, 2^52 is the largest pow2 that can be stored in a number primitive
```

Comment:
I suppose you'll be using a long on the C# side...

Reply:
`ulong` most likely, yeah. That's the 64-bit one, right?

---
In reply to: [266634369](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/269#discussion_r266634369) [](ancestors = 266634369)